analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -131,11 +133,34 @@ public bool HasMatchingScope(DiagnosticDescriptor descriptor)
             descriptor.CustomTags.Contains(tag);
     }
 
+    /// <summary>
+    /// Check if the current file path is included or excuded and caches the result.
+    /// </summary>
+    public bool ShouldAnalyzeFile(string filePath) =>
```

Comment:
This whole check should only happen in SonarLint context (`projectType == ProjectType.Unknown`)